### ISSUE TITLE
v 1.2.48 TypeError: Can not find node binding files from @swc/core-linux-x64-gnu

### ISSUE THREAD

**Issue by sporto** [swc-project/swc#1430]
**Describe the bug**

In 1.2.48 we are getting

```
TypeError: Can not find node binding files from @swc/core-linux-x64-gnu, @swc/core-linux-x64-gnux32, @swc/core-linux-x64-musl
```

This in Debian 10 (Buster)

1.2.47 was fine

**Config**

```json
{
  "jsc": {
    "parser": {
      "syntax": "typescript",
      "dynamicImport": true
    }
  }
}

```

**Expected behavior**

Compile without error

**Version**

The version of @swc/core: 1.2.48

**Additional context**

Adding @swc/core-linux-musl doesn't change this error

**Comment by kdy1** [comment]
cc @Brooooooklyn 
Do you have any idea?

**Comment by Brooooooklyn** [comment]
Seems like linux binary built on `ubuntu@20.04` and `Glibc` on it is not compatible with Node.js official docker images, I will fix the build CI and add some tests asap.

@kdy1 You can set  back the packages `@swc/core-{win32,linux,darwin,android}@1.2.47` to `latest` npm tag to avoid users to install the `1.2.48`.

**Comment by kdy1** [comment]
@Brooooooklyn Thanks!

**Comment by Kamadake** [comment]
I'd like to add on to this.

I'm running Ubuntu 18.04
`4.15.0-1027-gcp #28-Ubuntu SMP Tue Jan 15 12:29:48 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

and reverting the versions of @swc-core didn't do much (unless I don't know how to set back the package, I just changed the version to 1.2.47 in the pnpm-lock file)

my solution instead was to rebuild the swc.linux-x64-gnu.node. I cloned`@swc/core` and ran the build script then replaced the node file. it's a temporary solution but at least got us working again.

**Comment by garretteklof** [comment]
I'm running into this same issue on MacOS 11.2.3 and node:14 -- there are no binaries installed in @swc/core

**Comment by garretteklof** [comment]
I'm running into this same issue on MacOS `11.2.3` and `node:14` -- there are no binaries installed in `@swc/core`

**Comment by Brooooooklyn** [comment]
- Check if `node_modules/@swc/core-darwin-x64/swc.darwin-x64.node` existed 
- Check the ***System preferences*** -> ***Security & Privacy*** -> ***General*** tab, if `swc..darwin-x64.node` blocked by system.

@garretteklof

**Comment by garretteklof** [comment]
@Brooooooklyn sorry, I should've clarified. I'm running in Docker `swc...darwin-x64.node` existed (both host / container). The error inside of the Docker container was like the OP `TypeError: Can not find node binding files from @swc/core-linux-x64-gnu, @swc/core-linux-x64-gnux32, @swc/core-linux-x64-musl`

I realized it was because I was binding `node_modules` in docker-compose `volumes` 🤦 did not realize that `.dockerignore` has to be at the root (part of a monorepo). Anyway, thanks for the fast reply!

**Comment by ycjcl868** [comment]
+1 same problem

> ERROR: Can not find node binding files from @swc/core-linux-x64-gnu, @swc/core-linux-x64-gnux32, @swc/core-linux-x64-musl and

**Comment by Brooooooklyn** [comment]
> +1 same problem

Please provide your OS version, node version, and the stdout of `ls -R node_modules/@swc`

**Comment by ycjcl868** [comment]
> > +1 same problem
> 
> @ycjcl868 Please provide your OS version, node version, and the stdout of `ls -R node_modules/@swc`.

in Github Action

**Comment by Brooooooklyn** [comment]
Open https://github.com/napi-rs/node-rs/issues/427 to track this problem

**Comment by vyacheslavzhabitsky** [comment]
have the same issue on ubuntu, no ideas on how to fix

**Comment by mrcrowl** [comment]
I just encountered a very similar problem, using npm, where `package-lock.json` ends up only including the optional dependencies for the platform I'm on (macOS/darwin). 

To be specific, the `package-lock.json` has an entry for `node_modules/@swc/core` which has the full list of `optionalDependencies`, however, the modules for the dependencies _themselves_ are missing from the lock file.

When this reaches a build pipeline (in our case, BuildKite on amazon linux 2), it fails to install because `@swc/core-linux-x64-gnu` is missing.

Reproducable by doing the following: (npm v8.11.0, node v16.15.1, macOS 11.6):

1. Start on macOS, with any recent version of `@swc/core` already installed (and working for CI).
2. Delete the `package-lock.json`
3. Run `npm install`
4. A new `package-lock.json` is generated, but is missing most of the module entries for the optional dependencies (except for `@swc/core-darwin-x64`).

The fix is just to reinstall `@swc/core`.  i.e. `npm install @swc/core` 

After this, the missing modules for the optional dependencies are now included in `package-lock.json`.

**Comment by mrcrowl** [comment]
TL;DR `package-lock.json` is not correctly recreated for `@swc/core` when generated from scratch. Hope that helps!
